packages/dev/sharedUiComponents/src/nodeGraphSystem/graphCanvas.tsx

```diff
@@ -1248,6 +1248,20 @@ export class GraphCanvasComponent extends React.Component<IGraphCanvasComponentP
             const emittedNodeData = newDefaultInput.data;
 
             pointA = emittedNodeData.getPortByName(pointName)!;
+            if (!pointA) {
+                for (let i = 0; i < emittedNodeData.outputs.length; i++) {
+                    const output = emittedNodeData.outputs[i];
+                    const outputData = output.data;
+                    const outputBlockType = outputData._blockType;
+                    if (outputBlockType === pointB.data._blockType) {
+                        pointA = output;
+                        break;
+                    }
+                }
+            }
+            if (!pointA) {
+                return;
+            }

```

Comment:
This can move inside the last if. Otherwise, as an overview, it looks a bit off - you are checking for the same thing you checked in the last if statement